### PR TITLE
Code tweaks

### PR DESCRIPTION
**Pull request by miloskrca**
* Fix potential deadlock. A deadlock can occur when we have multiple hashes we are searching for and one of the searches fails
* Better error handling with more context information
* Use base 10 when parsing lastTweetIdValue
* Close settings and lastTweetId files
* Do not crash when lastTweetId file is not present (happens on first run)

### CODE REVIEW

File: tweetbot.go [hunk 1]
```diff
@@ -36,56 +38,93 @@ func check(e error) {
...
+
+			if lastTweetID != 0 {
+				params.SinceID = lastTweetID
+			}
+
+			search, _, err := client.Search.Tweets(params)
+			if err != nil {
+				log.Printf("could not search tweets for hash '%s': %s", hash, err)
+				// a search did not execute properly, send an empty object so we don't deadlock
+				out <- &twitter.Search{}
```
Comment (miloskrca) [inline tweetbot.go:125]: If we did not do this we would block indefinitely below while trying to read the "out" channel

File: tweetbot.go [hunk 2]
```diff
@@ -36,56 +38,93 @@ func check(e error) {
...
+	}
 
-	lastTweetId, err := ioutil.ReadAll(fileReference)
-	check(err)
+	var lastTweetID int64
+	if id, err := getLastTweetID(); err != nil {
+		log.Printf("could not get last tweet id: %s", err)
+	} else {
+		lastTweetID = id
+	}
```
Comment (miloskrca) [inline tweetbot.go:96]: If the lastTweetId file is not present everything would stop but ideally we continue and create the file after the first run

File: tweetbot.go [hunk 3]
```diff
@@ -36,56 +38,93 @@ func check(e error) {
...
 	if err != nil {
-		return err
+		return fmt.Errorf("could not open file: %s", err)
 	}
-	defer file.Close()
+	defer closeFile(file)
 
 	_, err = io.WriteString(file, data)
-	check(err)
-	return file.Sync()
```
Comment (miloskrca) [inline tweetbot.go:45]: the file.Sync is not necessary, just by Closing the content will be written to the file system

**Comment by tomasnorre** [comment]
Thanks for your input, will have a look as soon as I find a little free time :)

**Pull request merged**